File: Solutions/1470/1470.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2022/8/29 10:10
# File: 1470.py
# Desc: 

class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        ret = list()
        for i in range(n):
            ret.append(nums[i])
            ret.append(nums[i+n])
        return ret

File: Solutions/1470/1470.c
/***************************************************************************************************
 * Copyright: washing
 * FileName: 1470.c
 * Author: washing
 * Version: 1.0
 * Date: 2022-8-29 10:11:06
***************************************************************************************************/




/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* shuffle(int* nums, int numsSize, int n, int* returnSize){
    int *ret = (int *)malloc(sizeof(int) * numsSize);
    int idx = 0;
    for (int i=0; i<n; i++) {
        ret[idx++] = nums[i];
        ret[idx++] = nums[i+n];
    }
    *returnSize = numsSize;
    return ret;
}
